packages/core/src/workflows/workflow.ts

```diff
@@ -1139,6 +1148,15 @@ export class Workflow<
       });
     }
 
+    // Bind methods to ensure they work when destructured
+    run.start = run.start.bind(run);
+    run.resume = run.resume.bind(run);
+    run.stream = run.stream.bind(run);
+    run.cancel = run.cancel.bind(run);
+    run.sendEvent = run.sendEvent.bind(run);
+    run.watch = run.watch.bind(run);
+    run.getState = run.getState.bind(run);
```

Comment:
@wardpeet how about this? just binding in createRun and createRunAsync since it's not immediately clear those return a class instance